+ source ../common/common_lib.sh
++ source /root/mugen-riscv/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/mugen-riscv/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Thu Dec 29 06:38:57 2022 - INFO  - Start environmental preparation.
+ DNF_INSTALL rasdaemon
+ pkgs=rasdaemon
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py install --pkgs rasdaemon --node 1 --tempfile ''
+ tmpfile2=/tmp/tmposj_bgnv
+ '[' -z '' ']'
+ tmpfile=/tmp/tmposj_bgnv
+ service=ras-mc-ctl.service
++ date '+%Y-%m-%d %T'
+ log_time='2022-12-29 06:39:49'
+ flag=false
++ getenforce
++ grep Enforcing
+ '[' ']'
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Thu Dec 29 06:39:50 2022 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Thu Dec 29 06:39:50 2022 - INFO  - Start testing...
+ test_restart ras-mc-ctl.service
+ service=ras-mc-ctl.service
+ systemctl restart ras-mc-ctl.service
Job for ras-mc-ctl.service failed because the control process exited with error code.
See "systemctl status ras-mc-ctl.service" and "journalctl -xeu ras-mc-ctl.service" for details.
+ CHECK_RESULT 1 0 0 'ras-mc-ctl.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='ras-mc-ctl.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'ras-mc-ctl.service restart failed'
+ LOG_ERROR 'ras-mc-ctl.service restart failed'
+ message='ras-mc-ctl.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'ras-mc-ctl.service restart failed'
Thu Dec 29 06:39:52 2022 - ERROR - ras-mc-ctl.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 34'
+ message='../common/common_lib.sh line 34'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 34'
Thu Dec 29 06:39:53 2022 - ERROR - ../common/common_lib.sh line 34
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status ras-mc-ctl.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'ras-mc-ctl.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='ras-mc-ctl.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'ras-mc-ctl.service restart failed'
+ LOG_ERROR 'ras-mc-ctl.service restart failed'
+ message='ras-mc-ctl.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'ras-mc-ctl.service restart failed'
Thu Dec 29 06:39:59 2022 - ERROR - ras-mc-ctl.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 37'
+ message='../common/common_lib.sh line 37'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 37'
Thu Dec 29 06:40:00 2022 - ERROR - ../common/common_lib.sh line 37
+ return 0
+ systemctl stop ras-mc-ctl.service
+ CHECK_RESULT 0 0 0 'ras-mc-ctl.service stop failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='ras-mc-ctl.service stop failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status ras-mc-ctl.service
+ grep 'Active: inactive'
+ CHECK_RESULT 1 0 0 'ras-mc-ctl.service stop failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='ras-mc-ctl.service stop failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'ras-mc-ctl.service stop failed'
+ LOG_ERROR 'ras-mc-ctl.service stop failed'
+ message='ras-mc-ctl.service stop failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'ras-mc-ctl.service stop failed'
Thu Dec 29 06:40:06 2022 - ERROR - ras-mc-ctl.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 42'
+ message='../common/common_lib.sh line 42'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 42'
Thu Dec 29 06:40:07 2022 - ERROR - ../common/common_lib.sh line 42
+ return 0
+ systemctl start ras-mc-ctl.service
Job for ras-mc-ctl.service failed because the control process exited with error code.
See "systemctl status ras-mc-ctl.service" and "journalctl -xeu ras-mc-ctl.service" for details.
+ CHECK_RESULT 1 0 0 'ras-mc-ctl.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='ras-mc-ctl.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'ras-mc-ctl.service start failed'
+ LOG_ERROR 'ras-mc-ctl.service start failed'
+ message='ras-mc-ctl.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'ras-mc-ctl.service start failed'
Thu Dec 29 06:40:09 2022 - ERROR - ras-mc-ctl.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 44'
+ message='../common/common_lib.sh line 44'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 44'
Thu Dec 29 06:40:09 2022 - ERROR - ../common/common_lib.sh line 44
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status ras-mc-ctl.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'ras-mc-ctl.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='ras-mc-ctl.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'ras-mc-ctl.service start failed'
+ LOG_ERROR 'ras-mc-ctl.service start failed'
+ message='ras-mc-ctl.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'ras-mc-ctl.service start failed'
Thu Dec 29 06:40:16 2022 - ERROR - ras-mc-ctl.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 47'
+ message='../common/common_lib.sh line 47'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 47'
Thu Dec 29 06:40:17 2022 - ERROR - ../common/common_lib.sh line 47
+ return 0
+ test_enabled ras-mc-ctl.service
+ service=ras-mc-ctl.service
++ systemctl is-enabled ras-mc-ctl.service
+ state=disabled
+ '[' disabled == enabled ']'
+ '[' disabled == disabled ']'
++ systemctl enable ras-mc-ctl.service
++ grep 'Created symlink'
++ awk '{print $3}'
+ symlink_file=/etc/systemd/system/multi-user.target.wants/ras-mc-ctl.service
+ find /etc/systemd/system/multi-user.target.wants/ras-mc-ctl.service
/etc/systemd/system/multi-user.target.wants/ras-mc-ctl.service
+ CHECK_RESULT 0 0 0 'ras-mc-ctl.service enable failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='ras-mc-ctl.service enable failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl disable ras-mc-ctl.service
Removed /etc/systemd/system/multi-user.target.wants/ras-mc-ctl.service.
+ find /etc/systemd/system/multi-user.target.wants/ras-mc-ctl.service
find: ‘/etc/systemd/system/multi-user.target.wants/ras-mc-ctl.service’: No such file or directory
+ CHECK_RESULT 1 0 1 'ras-mc-ctl.service disable failed'
+ actual_result=1
+ expect_result=0
+ mode=1
+ error_log='ras-mc-ctl.service disable failed'
+ '[' -z 1 ']'
+ '[' 1 -eq 0 ']'
+ test 1x == 0x
+ return 0
+ grep -v -i 'DEBUG\|INFO\|WARNING'
+ grep -i 'fail\|error'
+ journalctl --since '2022-12-29 06:39:49' -u ras-mc-ctl.service
+ grep -v 'Error: No dimm labels'
Dec 29 06:39:51 openEuler-riscv64 ras-mc-ctl[3694]: failed to run : Broken pipe
Dec 29 06:39:51 openEuler-riscv64 systemd[1]: ras-mc-ctl.service: Failed with result 'exit-code'.
Dec 29 06:39:51 openEuler-riscv64 systemd[1]: Failed to start Initialize EDAC v3.0.0 Drivers For Machine Hardware.
Dec 29 06:40:08 openEuler-riscv64 ras-mc-ctl[3710]: failed to run : Broken pipe
Dec 29 06:40:08 openEuler-riscv64 systemd[1]: ras-mc-ctl.service: Failed with result 'exit-code'.
Dec 29 06:40:08 openEuler-riscv64 systemd[1]: Failed to start Initialize EDAC v3.0.0 Drivers For Machine Hardware.
+ CHECK_RESULT 0 0 1 'There is an error message for the log of ras-mc-ctl.service'
+ actual_result=0
+ expect_result=0
+ mode=1
+ error_log='There is an error message for the log of ras-mc-ctl.service'
+ '[' -z 0 ']'
+ '[' 1 -eq 0 ']'
+ test 0x == 0x
+ test -n 'There is an error message for the log of ras-mc-ctl.service'
+ LOG_ERROR 'There is an error message for the log of ras-mc-ctl.service'
+ message='There is an error message for the log of ras-mc-ctl.service'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'There is an error message for the log of ras-mc-ctl.service'
Thu Dec 29 06:40:26 2022 - ERROR - There is an error message for the log of ras-mc-ctl.service
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_ras-mc-ctl.sh line 40'
+ message='oe_test_service_ras-mc-ctl.sh line 40'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_ras-mc-ctl.sh line 40'
Thu Dec 29 06:40:26 2022 - ERROR - oe_test_service_ras-mc-ctl.sh line 40
+ return 0
+ test_reload ras-mc-ctl.service
+ service=ras-mc-ctl.service
+ systemctl start ras-mc-ctl.service
Job for ras-mc-ctl.service failed because the control process exited with error code.
See "systemctl status ras-mc-ctl.service" and "journalctl -xeu ras-mc-ctl.service" for details.
+ grep 'Job type reload is not applicable'
+ systemctl reload ras-mc-ctl.service
Failed to reload ras-mc-ctl.service: Job type reload is not applicable for unit ras-mc-ctl.service.
+ CHECK_RESULT 0 0 0 'Job type reload is not applicable for unit ras-mc-ctl.service'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Job type reload is not applicable for unit ras-mc-ctl.service'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl status ras-mc-ctl.service
+ grep 'Active: active'
+ systemctl status ras-mc-ctl.service
+ grep 'inactive (dead)'
+ return 1
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Thu Dec 29 06:40:28 2022 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 6
+ test 6 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Thu Dec 29 06:40:29 2022 - ERROR - failed to execute the case.
+ exit 6
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Thu Dec 29 06:40:30 2022 - INFO  - start environment cleanup.
+ systemctl stop ras-mc-ctl.service
+ '[' false = true ']'
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmposj_bgnv ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmposj_bgnv
+ '[' 0 -ne 0 ']'
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Thu Dec 29 06:40:51 2022 - INFO  - Finish environment cleanup!
